### PR TITLE
Adding output file support

### PR DESCRIPTION
**Pull request by troelsfr**
This PR adds support for output files through the new flag `--output [filename]`.

### CODE REVIEW

**Review by cesarzc — APPROVED** [review]

**Review by swernli — APPROVED** [review]
One minor feedback on the error code, otherwise the change looks good. Thanks for the quick turn around!

File: qir/qat/Apps/Qat/Qat.cpp [hunk 1]
```diff
@@ -296,13 +296,35 @@ int main(int argc, char** argv)
...
+                    fout << *module << "\n";
+                }
+                else
+                {
+                    llvm::WriteBitcodeToFile(*module, fout);
+                }
+
+                if (ec.value() != 0)
+                {
+                    throw std::runtime_error("Failed to write output to file.");
```
[the remark this answers is not in the document]
Comment (cesarzc) [reply qir/qat/Apps/Qat/Qat.cpp:326]: @troelsfr if QAT fails, would it still write details to stderr even if an output file is passed?

[the remark this answers is not in the document]
Comment (troelsfr) [reply qir/qat/Apps/Qat/Qat.cpp:326]: @swernli point taken - let's make that a subsequent PR. @cesarzc not if you use the save logs option.

[the remark this answers is not in the document]
Comment (cesarzc) [reply qir/qat/Apps/Qat/Qat.cpp:326]: So if the `--output` is used then either bitcode or error details are written to the file?

[the remark this answers is not in the document]
Comment (troelsfr) [reply qir/qat/Apps/Qat/Qat.cpp:326]: `--output` is just the output IR either as BC or LL. The format depends on whether you have `-S` or not.

**Pull request merged**